### PR TITLE
Fix silent query-axis mask/bias collapse in memory-efficient (FLASH) attention

### PR DESCRIPTION
**Pull request by qflen**
## Summary

`bias_fn` in `dot_product_attention_queries_per_head` (`tabfm/src/jax/memory_efficient_attention.py`) hardcoded `slice_q_len = 1`, with the correct `min()` slicing commented out one line above, so any bias/mask that varies along the query axis is collapsed to each chunk's first row.

## Why it matters

Query-varying masks are supported: the docstring bias shape carries a full `q_length`, the shape-check accepts it, and `MultiheadAttention`'s 4-D `[B, #N, #T, T_src]` contract pins only `#N` to 1. 

FLASH is the default backend for the column and ICL layers (`tabfm_v1_0_0.py`), and on such a mask it silently returns each chunk's first row while the JAX backend of the same layer is correct. A kv-broadcastable bias (last dim 1) also crashes `lax.dynamic_slice`.

## Reproduction

Query-varying mask `[B, 1, T, S]` (query `t` attends keys `<= t + 50`), `T = S = 256`, chunk 128. "first-row collapsed" is the same mask with each chunk forced to its first row, which is what the bug produces.

<details>
<summary>Full repro script</summary>

```python
import jax
import jax.numpy as jnp
import numpy as np
from tabfm.src.jax import memory_efficient_attention as mea

B, H, D, T, S, CHUNK = 2, 2, 8, 256, 256, 128
rng = np.random.default_rng(0)
q = jnp.asarray(rng.standard_normal((B, T, H, D)), jnp.float32)
k = jnp.asarray(rng.standard_normal((B, S, H, D)), jnp.float32)
v = jnp.asarray(rng.standard_normal((B, S, H, D)), jnp.float32)

qi = jnp.arange(T)[:, None]
ki = jnp.arange(S)[None, :]
# query-varying mask [B, 1, T, S]: query t attends keys <= t + 50
bias = jnp.where(
    jnp.broadcast_to((ki <= qi + 50)[None, None], (B, 1, T, S)), 0.0, -1e30)
# the same mask with each chunk collapsed to its first query row (the bug's effect)
row0 = (jnp.arange(T) // CHUNK) * CHUNK
collapsed = jnp.where(
    jnp.broadcast_to((ki <= row0[:, None] + 50)[None, None], (B, 1, T, S)),
    0.0, -1e30)

def exact(b):
  return jax.nn.dot_product_attention(query=q, key=k, value=v, bias=b, scale=1.0)

flash = jnp.asarray(mea.dot_product_attention_multihead(
    query=q, key=k, value=v, bias=bias, dtype=q.dtype,
    query_chunk_size=CHUNK, key_chunk_size=CHUNK))
ref, col = exact(bias), exact(collapsed)
amax = lambda x: float(jnp.max(jnp.abs(x)))

print(f"query-varying mask, B={B} H={H} T={T} S={S} chunk={CHUNK}")
print(f"  exact-output scale (max|JAX|)          = {amax(ref):.2f}")
print(f"  max|FLASH - exact(JAX)|                = {amax(flash - ref):.3f}")
print(f"  max|FLASH - exact(first-row collapsed)|= {amax(flash - col):.2e}")
```

</details>

**Before**

```
query-varying mask, B=2 H=2 T=256 S=256 chunk=128
  exact-output scale (max|JAX|)          = 2.61
  max|FLASH - exact(JAX)|                = 3.832
  max|FLASH - exact(first-row collapsed)|= 1.67e-06
```
**After**
```
query-varying mask, B=2 H=2 T=256 S=256 chunk=128
  exact-output scale (max|JAX|)          = 2.61
  max|FLASH - exact(JAX)|                = 0.000
  max|FLASH - exact(first-row collapsed)|= 3.83e+00
```

## No behavior change for the shipped model

The `[B, 1, 1, S]` key-padding mask the model builds today is constant along the query axis, so its output is bit-identical before and after.

### CODE REVIEW

**Comment by weihaokong** [comment]
Thanks! Could you add tabfm/src/jax/memory_efficient_attention_test.py to the collect_ignore list in conftest.py (right after line 42) so a jax-less pytest doesn't fail collection?

**Comment by qflen** [comment]
@weihaokong thanks, done.

**Review by weihaokong — APPROVED** [review]
